### DISCUSSION TITLE
Using next/image behind a proxy

### DISCUSSION THREAD

**Discussion by icew1nd** [vercel/next.js#18376] [Help]
Do anyone have a good idea on how to use the new next/image API when you're behind a very strict proxy?

Currently i add the proxy information to our fetch calls, but it seems that next/image fetches the image internally with no option parameters for proxies. 

Getting: 

```
{message:
   'request to https://media.***/card.png failed, reason: connect ECONNREFUSED 1.1.1.1:443',
  type: 'system',
  errno: 'ECONNREFUSED',
  code: 'ECONNREFUSED' }
```

**Comment by nicgutierrez** [comment]
Hey @icew1nd , did you figure something out?

**Comment by bravetheheat** [comment]
One idea that comes to mind is that you could use an API route to configure a proxy and then direct all the requests there.

**Reply by nicgutierrez** [reply]
I think the main problem is `next/image` .. I can get API requests to work fine by using https://www.npmjs.com/package/tunnel and creating a custom agent with proxy to pass to the client. 

But `next/image` has its own server side logic that makes a request to fetch the image and that's where the issue happens

**Comment by a12791602** [comment]
i'm facing the same question , how do i fix it , is there anyone could tell me

**Reply by nicgutierrez** [reply]
Hey @a12791602 

I've ended up updating the source code in `node_modules` to include my `tunnel` library so that the proxy can be used in `next/image` http requests. I'm aware this is not a great solution, but it allowed me to keep working on things until a proxy solution is baked in the core. 

Hope this helps!

**Reply by radoslawgrochowski** [reply]
@nicgutierrez @a12791602 
using global-agent (https://github.com/gajus/global-agent) with custom server worked for me flawlessly

**Reply by liuxiaofeng1981** [reply]
@radoslawgrochowski Could you pls share your setup if possible? That will be really helpful.
